Backend/app/dashboard/provider-dashboard/src/API/Fleet/Registration.hs

```diff
@@ -61,27 +75,34 @@ type FleetOwnerVerifyAPI =
 handler :: FlowServer API
 handler =
   fleetOwnerLogin
-    :<|> fleetOwnerVerfiy
+    :<|> fleetOwnerVerify
     :<|> fleetOwnerRegister
 
 fleetOwnerLogin :: DP.FleetOwnerLoginReq -> FlowHandler APISuccess
 fleetOwnerLogin req = withFlowHandlerAPI' $ do
+  runRequestValidation DP.validateInitiateLoginReq req
   let merchantShortId = ShortId req.merchantId :: ShortId DM.Merchant
   merchant <- QMerchant.findByShortId merchantShortId >>= fromMaybeM (MerchantDoesNotExist merchantShortId.getShortId)
   let enabled = not $ fromMaybe False merchant.requireAdminApprovalForFleetOnboarding
       checkedMerchantId = skipMerchantCityAccessCheck merchantShortId
-  mbPerson <- QP.findByMobileNumber req.mobileNumber req.mobileCountryCode
+
   unless (req.city `elem` merchant.supportedOperatingCities) $ throwError (InvalidRequest "Invalid request city is not supported by Merchant")
+  let req' = buildFleetOwnerRegisterReq req
+  merchantServerAccessCheck merchant
+  mbPerson <- QP.findByMobileNumber req.mobileNumber req.mobileCountryCode
+  fleetOwnerRole <- QRole.findByDashboardAccessType DRole.FLEET_OWNER >>= fromMaybeM (RoleNotFound $ show DRole.FLEET_OWNER)
   res <- Client.callDynamicOfferDriverAppFleetApi checkedMerchantId req.city (.registration.fleetOwnerLogin) (Just enabled) req
   case mbPerson of
-    Just _ -> pure Success
+    Just person -> do
+      logError $ "PersonId mismatch for fleet owner registered: dashboard: " <> show person.id <> "; bpp: " <> show res.personId
```

Comment:
Looks like almost impossible case. We can think about proper fix if receive this log message